drivers/adc/Kconfig.stm32

```diff
@@ -0,0 +1,25 @@
+# Kconfig - ADC configuration options
+
+#
+# Copyright (c) 2019 Intel Corporation
+# Copyright (c) 2019 Endre Karlson
+# Copyright (c) 2019 Song Qiang <songqiang1304521@gmail.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig ADC_STM32
+	bool "STM32 ADC driver"
+	depends on SOC_FAMILY_STM32
+	help
+		Enable the driver implementation for the stm32xx ADC
+
+if ADC_STM32
+
+config ADC_1
+	bool "ADC1"
+	default y
+	help
+		Enable ADC1
+
```

Comment:
Miss ADC_2 and _3, since available in the driver

Reply:
I'm actually thinking about removing them, because there are so many different product lines, and different amount of ADCs they have, and also some ADCs share interrupt vector, so there should be some code for identifying which instance calls the interrupt. I would like another PR to do these things because this PR can be merged so users of STM32 can have basic ADC support.
But I'm worrying what Geo said about the merge window, but I see zephyr's doc says

> As a general rule, if you miss the merge window for a given feature, the best thing to do is to wait for the next development cycle. (An occasional exception is made for drivers for previously unsupported hardware; if they do not touch any other in-tree code, they cannot cause regressions and should be safe to add at any time).

So if this passes all tests and clean, we should be able to merge this PR soon, right?

Reply:
This is up to @galak to settle on this. Though, I'm not quite in favor of it. Even it could not be a source of regression, it could be a source of bugs, and branch maturity is tightly controlled.
Besides, there are several PR that could pretend be merged as this PR (L1/G0/WB). There is no reason this one is privileged. 
So I think the paragraph you mention should actually be corrected.

Reply:
Understand, so I'll just wait. ;)